src/opm/output/eclipse/RestartIO.cpp

```diff
@@ -614,33 +623,45 @@ namespace {
         return vectors;
     }
 
-    template <class OutputVector>
+    template <class OutputVector, class OutputVectorInt>
     void writeSolutionVectors(const RestartValue&             value,
                               const std::vector<std::string>& vectors,
-                              const bool                      write_double,
-                              OutputVector&&                  writeVector)
+                              OutputVector                    writeVectorF,
+                              OutputVectorInt                 writeVectorI)
     {
         for (const auto& vector : vectors) {
-            writeVector(vector, value.solution.data<double>(vector), write_double);
+            value.solution.at(vector).visit(std::move(VisitorOverloadSet{
```

Comment:
I don't think you need `std::move()` here.  `VisitorOverloadSet {...}` will be an x-value in the call to `CellData::visit()` which template argument deduction will perfectly forward without the explicit `move()`.

Reply:
indeed. i had it removed, forgot to recommit.